services/QuillLMS/client/app/bundles/Staff/components/comprehension/regexRules/regexRulesIndex.tsx

```diff
@@ -0,0 +1,156 @@
+import * as React from "react";
+import { Link, RouteComponentProps } from 'react-router-dom'
+import { useQuery } from 'react-query';
+import { firstBy } from 'thenby';
+
+import { getPromptsIcons } from '../../../helpers/comprehension';
+import { ActivityRouteProps, RuleInterface, RegexRuleInterface } from '../../../interfaces/comprehensionInterfaces';
+import { BECAUSE, BUT, SO } from '../../../../../constants/comprehension';
+import { fetchRules } from '../../../utils/comprehension/ruleAPIs';
+import { fetchActivity } from '../../../utils/comprehension/activityAPIs';
+import { DataTable, Error, Spinner } from '../../../../Shared/index';
+
+const RegexRulesIndex: React.FC<RouteComponentProps<ActivityRouteProps>> = ({ match }) => {
+  const { params } = match;
+  const { activityId } = params;
+
+    // get cached activity data to pass to rule
+    const { data: activityData } = useQuery({
+      queryKey: [`activity-${activityId}`, activityId],
+      queryFn: fetchActivity
+    });
+
+  const { data: rulesBased1Data } = useQuery({
+    // cache rules data for updates
+    queryKey: [`rules-${activityId}`, activityId, null, 'rules-based-1'],
```

Comment:
Isn't this always going to be the same data as the other two? Also, I think we should change `rulesBased1` etc to be `sentenceStructureRules` or whatever - it's hard to tell what's going on when they're just numbers.

Reply:
So this is what the actual rule type is in the DB, this only fetches Sentence Structure regex rules-- the url endpoint ends up looking like this: `http://quill.org/api/v1/comprehension/rules?rule_type=rules-based-1`